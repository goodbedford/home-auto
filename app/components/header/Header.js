import React from "react";

const PropTypes = React.PropTypes;
const propTypes = {
  date: PropTypes.string.isRequired
};

const Header = (props) => {
  return (
    <header className="home-auto__header">
      <span>Home Automation by Goodbedford LLC</span>
      <span>{props.date}</span>
    </header>
  );
};

Header.propTypes = propTypes;

export default Header;
