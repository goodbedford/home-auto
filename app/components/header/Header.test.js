/* eslint-disable no-undef */

import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";

import Header from "./Header";
import renderer from "react-test-renderer";


describe("<Header />", () => {
  it("Header snapshot test", () => {
    const wrapper = shallow(<Header date="01111979" />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
  it("Header should have title and date", () => {
    const wrapper = shallow(<Header date="01111979" />);
    expect(wrapper.find("span").length).toBe(2);
  });
  it("Header should show date", () => {
    const wrapper = shallow(<Header date="01111979" />);
    expect (wrapper.find("header").childAt(1).text()).toBe("01111979");
  });
});
