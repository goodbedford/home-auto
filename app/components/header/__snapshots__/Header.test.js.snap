// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Header snapshot test 1`] = `
<header
  className="home-auto__header"
>
  <span>
    Home Automation by Goodbedford LLC
  </span>
  <span>
    01111979
  </span>
</header>
`;
